draft-ietf-taps-interface.md

```diff
@@ -1554,38 +1554,55 @@ Security parameters and callbacks are partitioned based on their place in the li
 of Connection establishment. Similar to Transport Properties, both parameters and callbacks
 are inherited during cloning (see {{groups}}).
 
-### Specifying Security Parameters on a Pre-Connection
-
-Common security parameters such as TLS ciphersuites are known to implementations. Clients should
-use common safe defaults for these values whenever possible. However, as discussed in
-{{?RFC8922}}, many transport security protocols require specific
+This document specifies an abstract API, which may appear to conflict with the need
+for security parameters to be unambiguous in order to reduce the chances
+of implementation errors. The Transport Services system SHOULD provide reasonable,
+secure defaults for each enumerated security parameter, such that users of the system
+only need to specify parameters required to establish a secure connection
+('server-certificate', 'client-certificate'). Specifying specific security parameters
+from enumerated values (e.g., specific ciphersuites) may constrain the Transport
+Protocols that can be selected during connection establishment.
+
+Except as noted below, as with the rest of the Transport Services API, exact names of parameters and/or
+values of enumerations (e.g., ciphersuites) used in the security parameters are system
+and implementation specific, and should be chosen to follow the principle of least
+surprise for users of the platform / language environment in question.
+
+### Specifying Security Parameters on a Preconnection
+
+Common security parameters such as TLS ciphersuites are known to implementations.
+Clients should use common safe defaults for these values whenever possible.
+However, as discussed in {{?RFC8922}}, many transport security protocols require specific
 security parameters and constraints from the client at the time of configuration and
-actively during a handshake. These configuration parameters need to be specified in the
-pre-connection phase and are created as follows:
+actively during a handshake.
+
+The configuration parameters need to be specified in the pre-connection phase
+and are created as follows:
 
 ~~~
 SecurityParameters := NewSecurityParameters()
 ~~~
 
 Security configuration parameters and sample usage follow:
 
-- Application-layer protocol negotiation (ALPN) values: used to indicate which application-layer protocols
-are negotiated by the security protocol layer. See {{!ALPN=RFC7301}} for definition of the ALPN field. Note that the Transport Services System can provide ALPN values automatically, based on
-the protocols being used, if not explicitly specified by the application.
+- A certificate bundle identifying the Local Endpoint, whether as a server
+  certificate or a client certificate. Note that if the private keys associated
+  with a bundle are not available, e.g., since they are stored in hardware
+  security modules (HSMs), handshake callbacks must be used. See below for details.
```

Comment:
```suggestion
- One or more certificate bundles identifying the Local Endpoint, whether as a server
  certificate or a client certificate. Multiple bundles may be provided to allow selection
  among different protocol stacks that may require differently formatted bundles. The form and format of the certificate bundle is implementation-specific. Note that if the private keys associated
  with a bundle are not available, e.g., since they are stored in hardware
  security modules (HSMs), handshake callbacks must be used. See below for details.
```